sympy/polys/domains/algebraicfield.py

```diff
@@ -316,15 +316,15 @@ def __init__(self, dom, *ext, alias=None):
         self.symbols = self.gens = (self.ext,)
         self.unit = self([dom(1), dom(0)])
 
-        self.zero = self.dtype.zero(self.mod.rep, dom)
-        self.one = self.dtype.one(self.mod.rep, dom)
+        self.zero = self.dtype.zero(self.mod.to_list(), dom)
+        self.one = self.dtype.one(self.mod.to_list(), dom)
 
         self._maximal_order = None
         self._discriminant = None
         self._nilradicals_mod_p = {}
 
     def new(self, element):
-        return self.dtype(element, self.mod.rep, self.dom)
+        return self.dtype(element, self.mod.to_list(), self.dom)
```

Comment:
If ANP was built directly on top of DMP then we could just pass `self.mod` here rather than `self.mod.to_list()`.